frontend/src/metabase/dashboard/selectors/selectors.js

```diff
@@ -155,21 +155,22 @@ const getIsParameterValuesEmpty = createSelector(
 
 export const getCanShowAutoApplyFiltersToast = createSelector(
   [
-    getDashboardId,
+    getDashboard,
     getAutoApplyFiltersToastDashboardId,
     getIsAutoApplyFilters,
     getIsSlowDashboard,
     getIsParameterValuesEmpty,
   ],
   (
-    dashboardId,
+    dashboard,
     toastDashboardId,
     isAutoApply,
     isSlowDashboard,
     isParameterValuesEmpty,
   ) => {
     return (
-      dashboardId !== toastDashboardId &&
+      dashboard.can_write &&
```

Comment:
This prevents the auto-apply filters toast from appearing for users without collection curate permission because the reason the toast exist is to prompt users to click (modify the dashboard), so if they can't modify the dashboard, there's no need to show this toast at all.

![image](https://github.com/metabase/metabase/assets/1937582/4653edbe-e78c-4510-ad63-5883736c4505)